File: src/test2.py
import z3
import math
import sympy as sp
from adapter import Input, PathCond, Adapter


def ex5b():
    a1, a2 = z3.Ints('a1 a2')
    inp = Input([a1], [5])
    bad_pathconds = [
        PathCond([a1 < 1]),
        PathCond([a1 == 5])
        # PathCond([a2 < 1]),
        # PathCond([a2 == 5]),
        # PathCond([a3 < 1]),
        # PathCond([a3 == 5])
    ]
    return inp, bad_pathconds


inp, bad_pathconds = ex5b()
Adapter(inp, bad_pathconds).doit()

File: src/test2.cpp
#include <iostream>
#include <z3++.h>
#include "adapter.h"

using namespace std;
using namespace z3;

int main() {
    context c;
    // Define variables/ constants
    expr a1 = c.int_const("a1");
    expr a2 = c.int_const("a2");
    expr_vector vars = expr_vector(c);
    vars.push_back(a1);

    // Define values
    expr_vector vals = expr_vector(c);
    vals.push_back(c.int_val(5));
    
    // Define Input
    Input inp = Input(vars, vals, &c);

    // Define bad conditions
    vector<PathCond> badPathConds;
    expr_vector conjs = expr_vector(c);
    conjs.push_back(a1 < c.int_val(1));
    badPathConds.push_back(PathCond(conjs, &c));
    conjs = expr_vector(c);
    conjs.push_back(a1 == c.int_val(5));
    badPathConds.push_back(PathCond(conjs, &c));

    // Execute
    Adapter adap = Adapter(inp, badPathConds, &c);
    adap.doit();
    return 0;
}
